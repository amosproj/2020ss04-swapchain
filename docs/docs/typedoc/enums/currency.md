---
id: "currency"
title: "Currency"
sidebar_label: "Currency"
---

[swapchain documentation](../globals.md) › [Currency](currency.md)

## Index

### Enumeration members

- [BTC](currency.md#btc)
- [BTS](currency.md#bts)

## Enumeration members

### BTC

• **BTC**: = "BTC"

_Defined in [components/util/enums.ts:10](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L10)_

_Defined in [components/forms/enums.ts:10](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L10)_

---

### BTS

• **BTS**: = "BTS"

_Defined in [components/util/enums.ts:11](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L11)_

_Defined in [components/forms/enums.ts:11](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L11)_
