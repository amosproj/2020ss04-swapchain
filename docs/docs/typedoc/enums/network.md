---
id: "network"
title: "Network"
sidebar_label: "Network"
---

[swapchain documentation](../globals.md) › [Network](network.md)

## Index

### Enumeration members

- [MAINNET](network.md#mainnet)
- [TESTNET](network.md#testnet)

## Enumeration members

### MAINNET

• **MAINNET**: = "mainnet"

_Defined in [components/util/enums.ts:15](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L15)_

_Defined in [components/forms/enums.ts:15](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L15)_

---

### TESTNET

• **TESTNET**: = "testnet"

_Defined in [components/util/enums.ts:16](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L16)_

_Defined in [components/forms/enums.ts:16](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L16)_
