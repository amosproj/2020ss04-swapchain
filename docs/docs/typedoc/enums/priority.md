---
id: "priority"
title: "Priority"
sidebar_label: "Priority"
---

[swapchain documentation](../globals.md) › [Priority](priority.md)

## Index

### Enumeration members

- [HIGH](priority.md#high)
- [LOW](priority.md#low)
- [MEDIUM](priority.md#medium)

## Enumeration members

### HIGH

• **HIGH**:

_Defined in [components/util/enums.ts:20](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L20)_

_Defined in [components/forms/enums.ts:20](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L20)_

---

### LOW

• **LOW**:

_Defined in [components/util/enums.ts:22](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L22)_

_Defined in [components/forms/enums.ts:22](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L22)_

---

### MEDIUM

• **MEDIUM**:

_Defined in [components/util/enums.ts:21](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L21)_

_Defined in [components/forms/enums.ts:21](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L21)_
