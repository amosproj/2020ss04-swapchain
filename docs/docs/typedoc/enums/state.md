---
id: "state"
title: "State"
sidebar_label: "State"
---

[swapchain documentation](../globals.md) › [State](state.md)

## Index

### Enumeration members

* [ERROR](state.md#error)
* [FAILURE](state.md#failure)
* [IDLE](state.md#idle)
* [RUNNING](state.md#running)
* [SUCCESS](state.md#success)

## Enumeration members

###  ERROR

• **ERROR**:

*Defined in [components/util/enums.ts:5](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L5)*

*Defined in [components/forms/enums.ts:5](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L5)*

___

###  FAILURE

• **FAILURE**:

*Defined in [components/util/enums.ts:6](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L6)*

*Defined in [components/forms/enums.ts:6](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L6)*

___

###  IDLE

• **IDLE**:

*Defined in [components/util/enums.ts:2](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L2)*

*Defined in [components/forms/enums.ts:2](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L2)*

___

###  RUNNING

• **RUNNING**:

*Defined in [components/util/enums.ts:3](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L3)*

*Defined in [components/forms/enums.ts:3](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L3)*

___

###  SUCCESS

• **SUCCESS**:

*Defined in [components/util/enums.ts:4](https://github.com/chronark/swapchain/blob/6beff0a/src/components/util/enums.ts#L4)*

*Defined in [components/forms/enums.ts:4](https://github.com/chronark/swapchain/blob/6beff0a/src/components/forms/enums.ts#L4)*
