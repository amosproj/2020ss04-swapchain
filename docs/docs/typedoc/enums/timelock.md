---
id: "timelock"
title: "Timelock"
sidebar_label: "Timelock"
---

[swapchain documentation](../index.md) › [Globals](../globals.md) › [Timelock](timelock.md)

## Index

### Enumeration members

- [LONG](timelock.md#long)
- [MEDIUM](timelock.md#medium)
- [SHORT](timelock.md#short)

## Enumeration members

### LONG

• **LONG**: = 20

Defined in components/forms/enums.ts:20

---

### MEDIUM

• **MEDIUM**: = 13

Defined in components/forms/enums.ts:21

---

### SHORT

• **SHORT**: = 6

Defined in components/forms/enums.ts:22
