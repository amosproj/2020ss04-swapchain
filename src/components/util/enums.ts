export enum State {
  IDLE,
  RUNNING,
  SUCCESS,
  ERROR,
  FAILURE,
}

export enum Currency {
  BTC = "BTC",
  BTS = "BTS",
}

export enum Network {
  MAINNET = "mainnet",
  TESTNET = "testnet",
}

export enum Priority {
  HIGH,
  MEDIUM,
  LOW,
}
