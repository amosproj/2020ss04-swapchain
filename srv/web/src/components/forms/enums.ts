export enum State {
  IDLE,
  INVALID_INPUT,
  RUNNING,
  SUCCESS,
  ERROR,
  FAILURE,
}

export enum Currency {
  BTC,
  BTS,
}

export enum Network {
  MAINNET = "mainnet",
  TESTNET = "testnet",
}

export enum Timelock {
  LONG,
  MEDIUM,
  SHORT,
}

export enum Priority {
  HIGH,
  MEDIUM,
  LOW,
}
