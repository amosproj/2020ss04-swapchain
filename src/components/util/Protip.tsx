import React from "react"
import { ReactComponent as LightBulb } from "../../icons/light-bulb.svg"
import { Network } from "../util/enums"

interface Props {
  network: Network
}

export const Protip = (props: Props) => {
  return (
    <div className="flex flex-col items-center px-3 py-2 space-x-1 md:flex-row">
      <div className="flex items-center">
        <LightBulb className="h-6 text-gray-600"></LightBulb>
        <span className="font-semibold text-gray-900">ProTip!</span>
      </div>
      <div className="flex flex-col items-center md:flex-row">
        <span className="text-center text-gray-700">Check your transaction on the blockchain:</span>
        <div className="flex px-2 space-x-2">
          <a
            target="blank"
            className="text-blue-400"
            href={props.network === Network.MAINNET ? "https://blockstream.info/" : "https://blockstream.info/testnet/"}
          >
            Bitcoin
          </a>
          <a target="blank" className="text-blue-400" href="https://wallet.bitshares.org/#/">
            Bitshares
          </a>
        </div>
      </div>
    </div>
  )
}
