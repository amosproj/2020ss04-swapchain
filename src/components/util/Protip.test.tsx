import React from "react"
import { Protip } from "./Protip"
import renderer from "react-test-renderer"
import { Network } from "./enums"

it("renders correctly for testnet", () => {
  const tree = renderer.create(<Protip network={Network.TESTNET}></Protip>).toJSON()
  expect(tree).toMatchSnapshot()
})

it("renders correctly for mainnet", () => {
  const tree = renderer.create(<Protip network={Network.MAINNET}></Protip>).toJSON()
  expect(tree).toMatchSnapshot()
})
