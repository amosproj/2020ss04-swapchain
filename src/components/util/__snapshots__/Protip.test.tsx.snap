// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for mainnet 1`] = `
<div
  className="flex flex-col items-center px-3 py-2 space-x-1 md:flex-row"
>
  <div
    className="flex items-center"
  >
    <svg
      className="h-6 text-gray-600"
    >
      light-bulb.svg
    </svg>
    <span
      className="font-semibold text-gray-900"
    >
      ProTip!
    </span>
  </div>
  <div
    className="flex flex-col items-center md:flex-row"
  >
    <span
      className="text-center text-gray-700"
    >
      Check your transaction on the blockchain:
    </span>
    <div
      className="flex px-2 space-x-2"
    >
      <a
        className="text-blue-400"
        href="https://blockstream.info/"
        target="blank"
      >
        Bitcoin
      </a>
      <a
        className="text-blue-400"
        href="https://wallet.bitshares.org/#/"
        target="blank"
      >
        Bitshares
      </a>
    </div>
  </div>
</div>
`;

exports[`renders correctly for testnet 1`] = `
<div
  className="flex flex-col items-center px-3 py-2 space-x-1 md:flex-row"
>
  <div
    className="flex items-center"
  >
    <svg
      className="h-6 text-gray-600"
    >
      light-bulb.svg
    </svg>
    <span
      className="font-semibold text-gray-900"
    >
      ProTip!
    </span>
  </div>
  <div
    className="flex flex-col items-center md:flex-row"
  >
    <span
      className="text-center text-gray-700"
    >
      Check your transaction on the blockchain:
    </span>
    <div
      className="flex px-2 space-x-2"
    >
      <a
        className="text-blue-400"
        href="https://blockstream.info/testnet/"
        target="blank"
      >
        Bitcoin
      </a>
      <a
        className="text-blue-400"
        href="https://wallet.bitshares.org/#/"
        target="blank"
      >
        Bitshares
      </a>
    </div>
  </div>
</div>
`;
